vamb/__main__.py

```diff
@@ -1430,6 +1427,14 @@ def run_bin_default(opt: BinDefaultOptions):
     del composition, abundance
     assert comp_metadata.nseqs == len(latent)
 
+    return latent
+
+
+def run_cluster_and_write_files(
+    latent, opt: BinDefaultOptions, composition: CompositionPath
+):
+    comp_metadata = composition.metadata
```

Comment:
Same here - you are conflating `Composition` and `CompositionPath`. The latter is a path to an existing `composition.npz` file created by Vamb.